src/hermes/commands/deposit/invenio.py

```diff
@@ -551,19 +551,39 @@ def _get_license_identifier(ctx: CodeMetaContext, license_api_url: str):
             "Licenses of type 'CreativeWork' are not supported."
         )
 
-    parsed_url = urlparse(license_url)
-    url_path = parsed_url.path.rstrip("/")
-    license_id = url_path.split("/")[-1]
-
+    # Fetch full list of licenses available... maybe we should cache this.
     response = requests.get(
-        f"{license_api_url}/{license_id}", headers={"User-Agent": hermes_user_agent}
+        f"{license_api_url}?size=1000", headers={"User-Agent": hermes_user_agent}
     )
-    if response.status_code == 404:
-        raise RuntimeError(f"Not a valid license identifier: {license_id}")
-    # Catch other problems
     response.raise_for_status()
+    valid_licenses = response.json()
+
+    def _search_license_info(_url):
+        for license_info in valid_licenses['hits']['hits']:
+            try:
+                if license_info['props']['url'] == _url:
+                    return license_info
+            except KeyError:
+                continue
+        else:
+            return None
+
+    license_info = _search_license_info(license_url)
+    if license_info is None and license_url.startswith('https://spdx.org/licenses/'):
+        response = requests.get(f"{license_url}.json", headers={"User-Agent": hermes_user_agent})
+        response.raise_for_status()
+
+        for license_cross_ref in response.json()['crossRef']:
+            if not license_cross_ref['isValid']:
+                continue
+
+            license_info = _search_license_info(license_cross_ref["url"])
+            if license_info is not None:
+                break
+        else:
+            raise RuntimeError(f"Could not resolve license URL {license_url} to a valid identifier.")
```

Comment:
**hint:** Yes, this is very ugly, but seems to be robust to me, even when InvenioRDM decides to use poper SPDX URLs.